passes/techmap/extract_reduce.cc

```diff
@@ -177,18 +216,19 @@ struct ExtractReducePass : public Pass
 					if (sig_to_sink[b[0]].size() + port_sigs.count(b[0]) == 1)
 					{
 						Cell* cell_b = sig_to_driver[b[0]];
-						if (((cell_b->type == "$_AND_" && gt == GateType::And) ||
-							(cell_b->type == "$_OR_" && gt == GateType::Or) ||
-							(cell_b->type == "$_XOR_" && gt == GateType::Xor)))
+						if (may_reduce(cell_b, gt))
 						{
-							// The cell here is the correct type, and it's definitely driving only
-							// this current cell.
-							bfs_queue.push_back(cell_b);
+							if (is_single_bit(cell_b, sigmap))
+							{
+								// The cell here is the correct type, and it's definitely driving only
+								// this current cell.
+								bfs_queue.push_back(cell_b);
+							}
 						}
 					}
 				}
 
-				log("  Cells:\n");
+				log("  Cells: count:%zu\n", cur_supercell.size());
```

Comment:
Please use `log("  Cells: count:%d\n", GetSize(cur_supercell));` instead. The `z` type specifier has portability issues.